cmd/loom/chainconfig/cmd.go

```diff
@@ -391,7 +396,133 @@ func GetValidatorInfoCmd() *cobra.Command {
 			return nil
 		},
 	}
-	cli.AddContractStaticCallFlags(cmd.Flags(), &flags)
+	cli.AddContractCallFlags(cmd.Flags(), &flags)
+	return cmd
+}
+
+const listPendingActionsCmdExample = `
+loom chain-cfg list-pending-actions
+`
+
+func ListPendingActionsCmd() *cobra.Command {
+	var flags cli.ContractCallFlags
+	cmd := &cobra.Command{
+		Use:     "list-pending-actions",
+		Short:   "show all pending actions to change setting",
+		Example: listPendingActionsCmdExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var resp cctype.ListPendingActionsResponse
+			err := cli.StaticCallContractWithFlags(
+				&flags, chainConfigContractName,
+				"ListPendingActions", &cctype.ListPendingActionsRequest{}, &resp,
+			)
+			if err != nil {
+				return err
+			}
+			out, err := formatJSON(&resp)
+			if err != nil {
+				return err
+			}
+			fmt.Println(out)
+			return nil
+		},
+	}
+	cli.AddContractCallFlags(cmd.Flags(), &flags)
+	return cmd
+}
+
+const chainConfigCmdExample = `
+loom chain-cfg config
+`
+
+func ChainConfigCmd() *cobra.Command {
+	var flags cli.ContractCallFlags
+	cmd := &cobra.Command{
+		Use:     "config",
+		Short:   "Get on-chain config",
+		Example: chainConfigCmdExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var resp cctype.ChainConfigResponse
+			err := cli.StaticCallContractWithFlags(
+				&flags, chainConfigContractName,
+				"ChainConfig", &cctype.ChainConfigRequest{}, &resp,
+			)
+			if err != nil {
+				return err
+			}
+			out, err := formatJSON(&resp)
+			if err != nil {
+				return err
+			}
+			fmt.Println(out)
+			return nil
+		},
+	}
+	cli.AddContractCallFlags(cmd.Flags(), &flags)
+	return cmd
+}
+
+const setSettingCmdExample = `
+loom chain-cfg set-setting AppStoreConfig.NumEvmKeysToPrune --value 100 --type uint64 --build 1200 -k private_key
+`
+
+func SetSettingCmd() *cobra.Command {
+	var flags cli.ContractCallFlags
+	var value string
+	var valueType string
+	var buildNumber uint64
+	cmd := &cobra.Command{
+		Use:     "set-setting <config name>",
+		Short:   "Set setting",
+		Example: setSettingCmdExample,
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if args[0] == "" || value == "" {
+				return fmt.Errorf("invalid config key")
+			}
+
+			// validate value
```

Comment:
Might be a good idea to validate the key name here too.

Reply:
Done